src/components/DateRangeInputWithFlowRunHistory.vue

```diff
@@ -0,0 +1,110 @@
+<template>
+  <DateRangeInput
+    v-model:startDate="internalStartDate"
+    v-model:endDate="internalEndDate"
+    v-model:viewingDate="viewingDate"
+    clearable
+  >
+    <template #date="{ date }">
+      <div class="date-range-input-with-flow-run-history__date">
+        <p>{{ date.getDate() }}</p>
+        <div class="date-range-input-with-flow-run-history__indicators">
+          <template v-for="history in getRunHistory(date)" :key="getRunHistoryKey(date, history.stateType)">
+            <div class="date-range-input-with-flow-run-history__indicator" :class="`bg-state-${history.stateType}-600`" />
+          </template>
+        </div>
+      </div>
+    </template>
+  </DateRangeInput>
+</template>
+
+<script lang="ts" setup>
+  import { useSubscriptionWithDependencies } from '@prefecthq/vue-compositions'
+  import { computed, ref, watch } from 'vue'
+  import DateRangeInput from '@/components/DateRangeInput.vue'
+  import { useFlowRunFilterFromRoute } from '@/compositions/useFlowRunFilterFromRoute'
+  import { useWorkspaceApi } from '@/compositions/useWorkspaceApi'
+  import { FlowRunHistoryMap } from '@/models/FlowRunHistoryMap'
+  import { StateHistory } from '@/models/StateHistory'
+  import { StateType } from '@/models/StateType'
+  import { FlowRunsHistoryFilter } from '@/types/UnionFilters'
+  import { secondsInDay } from '@/utilities/dates'
+  import { dateFunctions } from '@/utilities/timezone'
+
+  const api = useWorkspaceApi()
+
+  const { startDate, endDate, updateFilters, filter } = useFlowRunFilterFromRoute()
+
+  const internalStartDate = ref<Date | null>(startDate.value)
+  const internalEndDate = ref<Date | null>(endDate.value)
+  const viewingDate = ref<Date>()
+
+  watch([internalStartDate, internalEndDate], ([startDate, endDate]) => {
+    if (startDate && endDate) {
+      updateFilters({ startDate, endDate })
+    }
+  })
```

Comment:
This is such a nice example of watching an array of values, array destructuring, and object shorthand ❤️